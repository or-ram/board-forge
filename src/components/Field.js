import React, { Component } from 'react';
import * as P from 'prop-types';
import cx from 'classnames';

export default class Field extends Component {
  static propTypes = {
    col: P.number.isRequired,
    row: P.number.isRequired,
    onFieldClick: P.func,
    moveable: P.bool,
  };

  static defaultProps = {
    onFieldClick () {},
    moveable: false,
  };

  static isFieldDark ({ row, col }) {
    return Boolean(row % 2 ? (col + 1) % 2 : col % 2);
  }

  render () {
    const classNames = cx('field', {
      'field-dark': Field.isFieldDark(this.props),
      'field-light': !Field.isFieldDark(this.props),
      'field-moveable': this.props.moveable,
    });

    return (
      <div
        className={classNames}
        onClick={this.props.onFieldClick}
      >
        {this.props.children}
      </div>
    );
  }

  _colorClass () {
    return `field-${Field.isFieldDark(this.props) ? 'dark' : 'light'}`;
  }
}
