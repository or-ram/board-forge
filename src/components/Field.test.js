import React from 'react';
import Field from './Field';
import renderer from 'react-test-renderer';

describe('Field Test', () => {
  const component = renderer.create(
    <Field row={0} col={0} />,
  );
  let tree = null;

  beforeEach(() => {
    tree = component.toJSON();
  });

  test('Correctly rendered', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Dark class', () => {
    const darkTree = renderer.create(
      <Field row={0} col={1} />,
    ).toJSON();

    expect(darkTree.props.className).toMatch(/dark/);
  });

  test('Light class', () => {
    expect(tree.props.className).toMatch(/light/);
  });

  test('Moveable class', () => {
    const moveTree = renderer.create(
      <Field row={0} col={0} moveable={true} />,
    ).toJSON();

    expect(moveTree.props.className).toMatch(/moveable/);
  });

  describe('isFieldDark test', () => {
    test('coordinates 0 0 to be falsy', () => {
      expect(Field.isFieldDark({ row: 0, col: 0 })).toBeFalsy();
    });

    test('coordinates 0 1 to be truthy', () => {
      expect(Field.isFieldDark({ row: 0, col: 1 })).toBeTruthy();
    });

    test('coordinates 1 0 to be truthy', () => {
      expect(Field.isFieldDark({ row: 1, col: 0 })).toBeTruthy();
    });

    test('coordinates 1 1 to be falsy', () => {
      expect(Field.isFieldDark({ row: 1, col: 1 })).toBeFalsy();
    });
  });
});
