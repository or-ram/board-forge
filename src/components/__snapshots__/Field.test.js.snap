// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Test Correctly rendered 1`] = `
<div
  className="field field-light"
  onClick={[Function]}
/>
`;
